lib/widgets/message_list.dart

```diff
@@ -530,6 +534,68 @@ class MarkAsReadWidget extends StatelessWidget {
   }
 }
 
+class _TypingStatusState extends State<TypingStatusWidget> with PerAccountStoreAwareStateMixin<TypingStatusWidget> {
+  TypingStatus? model;
+
+  @override
+  void onNewStore() {
+    model?.removeListener(_modelChanged);
+    model = PerAccountStoreWidget.of(context).typingStatus
+      ..addListener(_modelChanged);
+  }
+
+  @override
+  void dispose() {
+    model?.removeListener(_modelChanged);
+    super.dispose();
+  }
+
+  void _modelChanged() {
+    setState(() {
+      // The actual state lives in [model].
+      // This method was called because that just changed.
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final narrow = widget.narrow;
+    const placeholder = SizedBox(height: 8);
```

Comment:
When the typing status is not shown, this puts an extra 8px between the last message and the mark-as-read button. If that's not part of the design, let's avoid it.

Reply:
Came across https://chat.zulip.org/#narrow/stream/48-mobile/topic/messages.20shift.20on.20mark-as-read, and I think the typing indicators have the same issue of causing a message shift. I wonder if we want to do something similar to #562.

Reply:
The follow-up discussion is here: https://chat.zulip.org/#narrow/stream/48-mobile/topic/messages.20shift.20on.20typing.20indicator/near/1925986